### PR TITLE
Create windows on mac

### PR DESCRIPTION
**Pull request by raphlinus**
Implement a subset of the window creation API. At this point, it
compiles and runs the examples in druid-shell, but with much
functionality stubbed out, and no attempt to run the higher level.

Uploading this as a checkpoint.

### CODE REVIEW

File: druid-shell/examples/perftest.rs [hunk 1]
```diff
@@ -26,9 +26,12 @@ use time::get_time;
 use kurbo::{Line, Rect};
 use piet::{FillRule, FontBuilder, RenderContext, Text, TextLayoutBuilder};
 
+#[cfg(target_os = "windows")]
+use druid_shell::platform::PresentStrategy;
```
Comment (rylev) [inline druid-shell/examples/perftest.rs]: Q: What is PresentStrategy?

File: druid-shell/src/mac/mod.rs [hunk 2]
```diff
@@ -13,8 +13,277 @@
...
+
+use util::assert_main_thread;
 
 #[derive(Clone, Default)]
-pub struct WindowHandle;
+pub struct WindowHandle {
+    /// This is an NSView, as our concept of "window" is more the top-level container holding
+    /// a view. Also, this is better for hosted applications such as VST.
+    ///
+    /// TODO: remove option (issue has been filed against objc, or we could manually impl default with nil)
```
Comment (rylev) [inline druid-shell/src/mac/mod.rs]: Can you link the issue number?

File: druid-shell/examples/perftest.rs (hunk 1 above)
Comment (raphlinus) [reply druid-shell/examples/perftest.rs]: See https://github.com/xi-editor/xi-win/issues/17 . It's part of my quest to make window resizing smooth on Windows. It's not something that's meaningful on other platforms, and most actual users will use the default. It's exposed for experimentation.

**Comment by raphlinus** [comment]
FWIW, I have an implementation of IdleHandler and anim (the latter is crude, based on continual invalidation, rather than CVDisplayLink, but is ok to get this into a working state) in my local repo. I can add to this PR or file a new one.

**Comment by rylev** [comment]
This looks good. Might be worth getting this in before adding the other stuff

**Pull request merged**